src/assets/scripts/client/navigationLibrary/FixCollection.js

```diff
@@ -162,8 +180,8 @@ class FixCollection extends BaseCollection {
      * @private
      */
     _buildFixModelsFromList(fixList, referencePosition) {
-        _forEach(fixList, (fixCoordinates, fixName) => {
-            const fixModel = new FixModel(fixName, fixCoordinates, referencePosition);
+        _forEach(fixList, (fixData) => {
+            const fixModel = new FixModel(fixData.name, fixData.spoken, fixData.coordinates, referencePosition); // FixModel accepts undefined spoken
```

Comment:
We should just pass `fixData` here, and have the FixModel constructor destruct the object in its _init() method. I can see you were just matching the pattern that was already there, but where practical we prefer to squish these together into an object and pass that object to the constructor. If anything is missing, we deal with it in ._init() and throw/return as appropriate.

Reply:
Sounds good!